src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2424,21 +2446,32 @@ async def detach(delegate):
         finally:
             await self.middleware.call('cache.pop', 'about_to_lock_dataset')
 
+        job.set_progress(80, f'Setting immutable flag on unmounted {id!r} dataset path')
+        cp = await run(['chattr', '+i', '-RV', ds['mountpoint']], check=False)
+        if cp.returncode:
+            raise CallError(
+                f'Unable to set immutable flag on unmounted {id!r} dataset\'s path '
+                f'({ds["mountpoint"]!r}): {cp.stderr.decode()}'
+            )
+
         await self.middleware.call_hook('dataset.post_lock', id)
+        job.set_progress(100, f'Successfully locked {id!r} dataset')
```

Comment:
Is this necessary every time we successfully end a job to set its progress to 100% and invent another success message?